libnmstate/nm/applier.py

```diff
@@ -70,6 +70,9 @@ def prepare_edited_ifaces_configuration(ifaces_desired_state):
         cur_con_profile = None
         if nmdev:
             cur_con_profile = connection.get_device_connection(nmdev)
+            if not cur_con_profile.get_interface_name():
+                set_conn = cur_con_profile.get_setting_connection()
```

Comment:
Will `get_setting_connection()' always return a connection setting? Otherwise it might also need to be created. Another issue is that this might change unspecific connection profiles (not bound to an interface) that happen to be active for an interface to be specific for this interface. I am not sure if this is what nmstate should do or if nmstate should then copy the original profile and store it with the new settings and the interface name.

Reply:
- Are you sure this is happening when the connection is in "activated" state? See #120 
- Why not update the new_con_profile with the interface if it is missing it? It sound safer to modify a new one over one which has been "read".